File: year2/semester2/MPP/Java Client Server Sockets/SablonJavaDB/src/main/java/me/teledonServices/ITeledonServices.java
package me.teledonServices;

import me.entities.Angajat;
import me.entities.Donator;
import me.networking.dto.AngajatDTO;
import me.networking.dto.CazCaritabilDTO;
import me.entities.Donatie;
import me.networking.dto.DonatieDTO;

import java.util.List;

public interface ITeledonServices {
    void login(Angajat angajat, ITeledonObserver client) throws TeledonExeption;
    //void donatieS(Integer cazId, String numeDonator, String adresa, String nrTel, Float sumaDonata) throws TeledonExeption;
    //void donatieS(Donatie donatie) throws TeledonExeption;
    //void donatieS(DonatieDTO donatie) throws TeledonExeption;
    void donatieS(Float sumaDonata, Integer idDonator, Integer cazId) throws TeledonExeption;
    void updateDonator(Donator donator) throws TeledonExeption;
    void saveDonator(Donator donator) throws TeledonExeption;
    void commitDonatori(String nume, String nrTel, String adresa) throws TeledonExeption;
    void check(Angajat angajat) throws TeledonExeption;
    Donator getDonator(String nume) throws TeledonExeption;
    void logout(Angajat angajat, ITeledonObserver client) throws TeledonExeption;
    List<CazCaritabilDTO> getCauriDTO() throws TeledonExeption;
    List<String> getDonators() throws TeledonExeption;
    List<AngajatDTO> getAngajati() throws TeledonExeption;
    List<String> getListaNumeAngajati() throws TeledonExeption;
    void updateAngajat(AngajatDTO a, String parola) throws TeledonExeption;
    Integer getCazId(CazCaritabilDTO cazCaritabilDTO) throws TeledonExeption;
}

File: year2/semester2/MPP/C# Client Server Sockets/Proiect_MPP_C#_VARIANTA_FINALA/servicess/ITeledonServices.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Domain_Repo.entities;
using DomainRepoLibrary.entities;

namespace servicess
{
    public interface ITeledonServices
    {
        void login(Angajat angajat, ITeledonObserver client);
        void donatieS(float sumaDonata, Int32 idDonator, Int32 cazId);
        void updateDonator(Donator donator);
        void saveDonator(Donator donator);
        void commitDonatori(string nume, string nrTel, string adresa);
        void check(Angajat angajat);
        Donator getDonator(string nume);
        void logout(Angajat angajat, ITeledonObserver client);
        List<CazDTO> getCauriDTO();
        List<string> getDonators();
        
        Int32 getCazId(CazDTO cazCaritabilDTO);
    }
}
